diff --git a/library/prolog_pack.pl b/library/prolog_pack.pl
index 2bd478466..51f217f2a 100644
--- a/library/prolog_pack.pl
+++ b/library/prolog_pack.pl
@@ -893,7 +893,8 @@ pack_unpack_from_local(Source0, PackTopDir, Name, PackDir, Options) :-
             link_file(RelPath, PackDir, symbolic),
             assertion(same_file(Source, PackDir))
         )
-    ;   is_git_directory(Source)
+    ;   \+ option(git(false), Options),
+        is_git_directory(Source)
     ->  remove_existing_pack(PackDir, Options),
         run_process(path(git), [clone, Source, PackDir], [])
     ;   prepare_pack_dir(PackDir, Options),
